FirebaseRemoteConfig/Sources/RCNConfigRealtime.m

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseRemoteConfig/Sources/RCNConfigRealtime.h"
+#import <Foundation/Foundation.h>
+
+NSInteger listenerCount = 1;
+
+@implementation FIRConfigUpdateListenerRegistration {
+  RCNConfigRealtime *_realtimeClient;
+  NSString *listenerName;
+}
+
+- (instancetype)initWithClass:(RCNConfigRealtime *)realtimeClient
+                 listenerName:(NSString *)listenerName {
+  self = [super init];
+  if (self) {
+    _realtimeClient = realtimeClient;
+    listenerName = listenerName;
+  }
+  return self;
+}
+
+- (void)remove {
+  [self->_realtimeClient removeConfigUpdateListener:listenerName];
+}
+
+@end
+
+@implementation RCNConfigRealtime
+
+- (instancetype)initWithClass {
+  self = [super init];
+  if (self) {
+    _listeners = [NSMutableDictionary alloc];
+  }
+
+  return self;
+}
+
+- (void)beginRealtimeStream {
+}
+
+- (void)pauseRealtimeStream {
+}
+
+- (FIRConfigUpdateListenerRegistration *)addConfigUpdateListener:
+    (FIRConfigUpdateCompletion)completionHandler {
+  NSString *listenerName = [@(listenerCount) stringValue];
+  [_listeners setValue:completionHandler forKey:listenerName];
```

Comment:
I wonder if you need a serial queue to manage this dictionary to ensure to conflicts, you can refer to other config source code to see how it was implemented: https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseRemoteConfig/Sources/FIRRemoteConfig.m#L152

It's better to have a separate serial queue that is only managing this class.

Reply:
Wrapped dictinary calls with serial queue